crates/aiken-lang/src/uplc.rs

```diff
@@ -102,15 +105,75 @@ impl<'a> CodeGenerator<'a> {
         }
 
         // Wrap the validator body if ifThenElse term unit error
-        term = if wrap_as_validator {
-            final_wrapper(term)
-        } else {
-            term
-        };
+        term = final_wrapper(term);
+
+        term = wrap_validator_args(term, &fun.arguments);
```

Comment:
shouldn't we only do that conditionally? I'd expect this entire feature to be a zero-cost abstraction for validators with a single function. We should only wrap things when necessary.

Reply:
I thought of that but it seemed easier to have a separate one that just received the whole validator. But yea we could switch back to one function.

Reply:
I am just thinking that it's less code to maintain and there's nothing really justifying having two functions here IMO 🤔

Reply:
No problem, we can make it work with just one. Original API was `generate(&TypedExpr, &[TypedArgs], bool)`

We can instead do

`generate(&TypedFunction, Option<&TypedFunction>, bool)`

But then in the test hint generate calls we need to construct fake Function structures instead of passing in body and empty args. Which I thought was a little ugly.